src/Client/Connection.cpp

```diff
@@ -197,20 +197,59 @@ void Connection::connect(const ConnectionTimeouts & timeouts)
                 , tcp_keep_alive_timeout_in_sec);
         }
 
-        in = std::make_shared<ReadBufferFromPocoSocket>(*socket);
+        in = std::make_shared<ReadBufferFromPocoSocketChunked>(*socket);
         in->setAsyncCallback(async_callback);
 
-        out = std::make_shared<WriteBufferFromPocoSocket>(*socket);
+        out = std::make_shared<WriteBufferFromPocoSocketChunked>(*socket);
         out->setAsyncCallback(async_callback);
         connected = true;
         setDescription();
 
         sendHello();
         receiveHello(timeouts.handshake_timeout);
 
+        bool out_chunked = false;
+        bool in_chunked = false;
+
+        if (server_revision >= DBMS_MIN_PROTOCOL_VERSION_WITH_CHUNKED_PACKETS)
+        {
+            auto is_chunked = [](const String & chunked_srv_str, const String & chunked_cl_str, const String & direction)
+            {
+                bool chunked_srv = chunked_srv_str.starts_with("chunked");
+                bool optional_srv = chunked_srv_str.ends_with("_optional");
+                bool chunked_cl = chunked_cl_str.starts_with("chunked");
+                bool optional_cl = chunked_cl_str.ends_with("_optional");
```

Comment:
Is there a reason to use strings like `chunked`/`optional` and write/read them instead of some binary encoding that will reqire 1-2 bytes?
Also it will be good to define and explain somewhere all possible combinations that can appear in this proto and what they mean (maybe use some enum with strings/defined constants, and use it instead of just strings)